src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -197,62 +211,97 @@ private Task ProcessGetId(HttpContext context)
+            // Collect the message and write it to the channel
+            // TODO: Need to use some kind of pooled memory here.
+            byte[] buffer;
+            using (var stream = new MemoryStream())
             {
-                // Collect the message and write it to the channel
-                // TODO: Need to use some kind of pooled memory here.
-                byte[] buffer;
-                using (var stream = new MemoryStream())
-                {
-                    await context.Request.Body.CopyToAsync(stream);
-                    buffer = stream.ToArray();
-                }
+                await context.Request.Body.CopyToAsync(stream);
+                buffer = stream.ToArray();
+            }
 
-                var format =
-                    string.Equals(context.Request.Query["format"], "binary", StringComparison.OrdinalIgnoreCase)
-                        ? Format.Binary
-                        : Format.Text;
+            var format =
+                string.Equals(context.Request.Query["format"], "binary", StringComparison.OrdinalIgnoreCase)
+                    ? Format.Binary
+                    : Format.Text;
 
-                var message = new Message(
-                    ReadableBuffer.Create(buffer).Preserve(),
-                    format,
-                    endOfMessage: true);
+            var message = new Message(
+                ReadableBuffer.Create(buffer).Preserve(),
+                format,
+                endOfMessage: true);
 
-                // REVIEW: Do we want to return a specific status code here if the connection has ended?
-                while (await state.Application.Output.WaitToWriteAsync())
+            // REVIEW: Do we want to return a specific status code here if the connection has ended?
+            while (await state.Application.Output.WaitToWriteAsync())
+            {
+                if (state.Application.Output.TryWrite(message))
                 {
-                    if (state.Application.Output.TryWrite(message))
-                    {
-                        break;
-                    }
+                    break;
                 }
             }
-            else
+        }
+
+        private void EnsureConnectionState(ConnectionState connectionState, HttpContext context, string transportName)
+        {
+            connectionState.Connection.User = context.User;
+
+            var transport = connectionState.Connection.Metadata.Get<string>("transport");
+            if (string.IsNullOrEmpty(transport))
+            {
+                connectionState.Connection.Metadata["transport"] = transportName;
+            }
+            else if (!string.Equals(transport, transportName, StringComparison.Ordinal))
+            {
+                throw new InvalidOperationException("Cannot change transports mid-connection");
+            }
+        }
+
+        private async Task<ConnectionState> GetConnectionAsync(HttpContext context)
```

Comment:
It was much more copy-pasta because we wrote the same response in numerous places.